pkg/util/resources_test.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2024 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"reflect"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// TestParseAPIGroupsString tests the TestParseAPIGroupsString function
+func TestParseAPIGroupsString(t *testing.T) {
+	// Define some test cases with inputs and expected outputs
+	testCases := []struct {
+		name     string
+		input    string
+		expected sets.Set[string]
+	}{
+		{
+			name:     "valid input with api group",
+			input:    "apps, networking.k8s.io, policy",
+			expected: sets.New("apps", "networking.k8s.io", "policy"),
```

Comment:
How does this work, with spaces no longer trimmed?

Reply:
you are right, fixed